feat: Connect CW Live Tail to Lambda functions

valerena opened the pull request:
Allow to "Tail Log Group" from Lambda functions, which will get the configured Log Group for a particular function and start a Live Tail session in that Log Group.

Because of the new version of the V2 SDK, we can now remove an extra code where we got a function's log group using V3.

## Problem

- For Lambda customers to use CloudWatch Live Tail, they need to know and find their Log group name inside the AWS Explorer, which can be hard to find if there are too many (or customers might not know which log group is configured in the first place, when they're using LoggingConfig). 


## Solution

Add a "Tail Log group" entry point to Lambda Functions, so Lambda customers can start Live Tail sessions directly from their Lambda functions, both in the AWS Explorer and in the recently added App Builder.

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be squash-merged at release time.
- Your code changes must meet the guidelines in [CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

valerena committed:
feat(lambda): Connect CW Live Tail to Lambda functions

Allow to "Tail Log Group" from Lambda functions, which will get the
configured Log Group for a particular function and start a Live Tail
session in that Log Group.

Because of the recently updated version of the SDK v2, we can take
the configured log group directly from the function's configuration
and we don't need an extra call using SDK V3 that was being made in
deployedNode.ts

chungjac committed:
numberOfTestsGenerated is correctly emitted as 0

ashishrp-aws committed:
Merge pull request #6346 from chungjac/fix-test-number-discrepancy

telemetry(amazonq): numberOfTestsGenerated correctly emits as 0

spfink committed:
fix(amazonq): Prevent customization override if user has manually selected a customization

## Problem

If a user was part of an AB group with a customization override, they could only ever use the customization from the override regardless of personal choice. 

## Solution

Changes the logic for when a customization will be overridden if the user is part of an AB group that has a customization override. With this change, if the user has manually selected a customization, it will not be overridden by the AB customization.

avi-alpert committed:
fix(amazonq): add error message for README update too large (#6347)

## Problem
If a request to update a README results in an updated README that
exceeds the size limit, a generic "I'm sorry, I ran into an issue
updating your README" error is displayed.

## Solution
Display a more specific error message when an updated README Exceeds the
size limit

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

hayemaxi committed:
revert: Q new user view badge handler (#6350)

Reverts
[259ca31](https://github.com/aws/aws-toolkit-vscode/commit/259ca31b827690ee3d96e4b44550ac8e4045b0c6#diff-53367fa008dcbcfc4cc4e6a9c97bae43ea889adfdec82209a996e7228cdf2c72)

Its purpose was to nudge inline completion users to the new chat
feature. The codepath is now no longer possible in new versions.

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

chungjac committed:
add message after accept/reject action

chungjac committed:
remove redundant sessions varaible

nkomonen-amazon committed:
fix(e2e): failing Auth Lambda (#6334)

## Problem

The CC tests are failing in the auth lambda step, saying that the
`user_code` value is not a string

## Solution

See code for solution

NOTE: This part is now fixed, but the auth lambda itself has its own
separate issue that needs to first be fixed

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

Signed-off-by: nkomonen-amazon <nkomonen@amazon.com>

muaydin committed:
fix(wizard): wizard remains in "not ready" state if a custom init() was provided #6339

## Problem
The wizard remained in a "not ready" state if a custom `init()` was
provided, requiring manual toggles in the test code.

## Solution
- Wrap the user’s init() with a small decorator that sets _ready = true.
- Remove the manual toggles in wizardTestUtils.ts, letting the wizard
handle readiness on its own.

dhasani23 committed:
fix(amazonq): broaden search strings (#6357)

## Problem

We search for 4 strings to indicate/infer Oracle SQL is present,
including `oracle.jdbc.OracleDriver`, but legacy software can instead
contain `oracle.jdbc.driver.OracleDriver`.

## Solution

Broaden/loosen the validation/search.

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

Co-authored-by: David Hasani <davhasan@amazon.com>

dhasani23 committed:
fix(amazonq): add link to docs (#6355)

## Problem

Link to docs could be helpful in this intro message.

## Solution

Add it.

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

Co-authored-by: David Hasani <davhasan@amazon.com>

chungjac committed:
Merge pull request #6354 from chungjac/remove-session

telemetry(amazonq): remove redundant session varaible

leigaol committed:
fix(amazonq): max 20k char for inline supplementalContext (#6366)

## Problem
In some rare cases, the total supplementalContext length exceeds 20k
character. With recent service side changes, we need to make sure the
total supplementalContext length do not exceed 20k in any situation.

## Solution
Make sure the total supplementalContext length do not exceed 20k in any
situation. Drop when necessary.

This is not customer facing so no change log is attached.


---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

spfink committed:
feat(amazonq): capability to send new context commands to AB groups (#6239)

## Problem

When working on new features available to AB groups, the new action are
not easily discoverable to users

## Solution

Add the ability to send a new command that will appear when a user types
@ in the chat window

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).

License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

Will-ShaoHua committed:
feat(inline completion): enhance inline completion context fetching (#6356)

## Problem
corresponding JB change
https://github.com/aws/aws-toolkit-jetbrains/pull/5256 (vscode logics
has been implemented)



## Solution


---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

Will-ShaoHua committed:
refactor(inline suggestion): camel case supplemental context strategy ids #6369

house cleaning and unify naming pattern of this field

corresponding JB commit
https://github.com/aws/aws-toolkit-jetbrains/pull/5256/commits/39bc048f2f437213bd25f7b4d32d53cbca9c9988

chungjac committed:
message now only in endSession

hayemaxi committed:
fix(dev): Q developer menu opening wrong auth state #6351

It's opening the toolkit global state value instead.

muaydin committed:
build: normalize filepath slashes, eol consistency #6353

## Problem
- Tests were failing or behaving inconsistently on different operating
systems due to improper file path normalization.
- End-of-line (EOL) conflicts were causing unnecessary formatting diffs
in collaborative workflows.

## Solution
- Updated the testFile variable in the test runner to normalize paths by
replacing backslashes (\) with forward slashes, ensuring cross-platform
compatibility.
- Added the Prettier rule 'prettier/prettier': ['error', { endOfLine:
'auto' }] to align EOL behavior with the host environment, avoiding
platform-specific conflicts.

siakmun-aws committed:
fix(amazonq): use server-side data to decide remaining code generations #6370

## Problem
Previously remaining code generation iterations are always displayed.

## Solution
Only display when it goes down to 2 or less.

Hweinstock committed:
ci(jscpd): improve output format #6373

## Problem
The output format of clones found in `jscpd` can be difficult to parse.
As an example, see
https://github.com/aws/aws-toolkit-vscode/actions/runs/12778633640/job/35621822359?pr=6370.
Additionally, it mentions file names and line numbers, but doesn't
provide a way to get to those files.

## Solution
- reduce friction by linking directly to the duplicates

avi-alpert committed:
test(amazonq): "edit README" E2E test case for /doc #6317

Add E2E test case for /doc: Making a specific change to the README

Hweinstock committed:
build(lint): add rule to prefer direct imports over index.ts #6372

## Problem
Importing from `..` or an `index.ts` file can lead to circular
dependencies.

## Solution
- add a lint rule to discourage importing from `..`. 
- migrate existing cases to import directly from the target module.

jpinkney-aws committed:
test(amazonq): Add e2e tests for general amazon q chat panel (#6279)

## Problem
- We want tests for the general amazon q panel
- We want other teams to quickly be able to write tests for their agents

## Solution
- Add general tests for the amazon q chat panel
- Add a template that other teams can use to write their tests
- Fix an issue that occurred only in tests where tab id's weren't
defined for the help message. The problem was that in the framework
everything is instant which meant a tab id was defined before that code
ran and was never passed through to processQuickActionCommand
- By default don't show the welcome page for now tests

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

nkomonen-amazon committed:
fix(sso): login with custom startUrl not allowed (#6368)

## Problem:

A user reported that a non-standard start url is technically
valid. This is because it can redirect to the underlying valid
start url that matches the pattern: https://xxxxxxxx.awsapps.com/start

## Solution:

Allow any URL, but warn users if they are using a non-standard one.
We will show a yellow warning message in this case.
The red error message is still shown when the input does not match a
URL in general.

## Examples 

### Invalid URL
<img width="315" alt="Screenshot 2025-01-14 at 4 33 58 PM"
src="https://github.com/user-attachments/assets/a5b2cb8a-c4fc-4678-a711-2f3f00bbe084"
/>

### Possibly valid since it may redirect to a valid URL
<img width="302" alt="Screenshot 2025-01-14 at 4 34 13 PM"
src="https://github.com/user-attachments/assets/0690f818-f4ba-4eae-b037-f856f5a2b2a0"
/>

### Missing the trailing `/start`
<img width="295" alt="Screenshot 2025-01-14 at 4 34 29 PM"
src="https://github.com/user-attachments/assets/8bcf3a4b-eba3-4bd8-8c68-24b709ee854d"
/>

### URL that also matches expected pattern
<img width="286" alt="Screenshot 2025-01-14 at 4 34 35 PM"
src="https://github.com/user-attachments/assets/eea2f2cb-6500-469c-9836-96ffc9cb5794"
/>

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

---------

Signed-off-by: nkomonen-amazon <nkomonen@amazon.com>

ashishrp-aws committed:
Merge pull request #6365 from chungjac/test-gen-completed-message

fix(amazonq): add message after accept/reject action

ashishrp-aws committed:
fix(amazonq): unsupported languages test generation query (#6363)

## Problem
- The test generation for unsupported language files and external files
is unreliable and flaky.
- Telemetry updated to include new metadata for existing metadata.

## Solution
- Fix to add file name as fallback in case of empty prompt or no code
selection.
- Telemetry updated and values passed to it in controller and messenger.
Essentially same PR :
https://github.com/aws/aws-toolkit-vscode/pull/6323 was closed last week
due to build issues.
---
Before:

https://github.com/user-attachments/assets/90ed6629-4753-41e6-9d11-b6d4dd0d6e7c

After fix:


https://github.com/user-attachments/assets/d0993345-cd2c-47ac-a8b5-1e9badf8fbac




- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

---------

Co-authored-by: Justin M. Keyes <jmkeyes@amazon.com>

Commit:
Release 1.43.0

Commit:
Release 3.42.0

Commit:
Update version to snapshot version: 1.44.0-SNAPSHOT

Commit:
Update version to snapshot version: 3.43.0-SNAPSHOT

Commit:
Merge release into master

leigaol committed:
telemetry(amazonq): calculate % of non-generated (user-written) code #5991

## Problem

With the release of many Q features(Inline Suggestion, chat, inline
chat, /dev, /test, /doc, /review, /transform), we need to know the %
code written by all Q features. This requires calculating and reporting
the user written code. The reporting of the code contribution of each Q
features was already implemented.


## Solution

Calculate and report the user written code for each language by
listening to document change events while Q is not making changes to the
editor.

We add flags to know whether Q is making temporary changes for
suggestion rendering or Q suggestion is accepted, by doing so, the
document change events are coming from the user.

We ignore certain document changes when their length of new characters
exceeds 50. Previous data driven research has shown that user tend to
copy a huge file from one place to another, making the user written code
count skyrocketing but that is actually some existing code not written
by the user.

We plan to first collect data from IDEs and let it run in the background
in shadow mode before we finish the service side aggregation, fix
possible bugs and eventually present the AI code written % to the
customers.

Note: The JB PR https://github.com/aws/aws-toolkit-jetbrains/pull/5215.
The JB implementation depends on a reliable JB internal message bus to
pass information. Using VSC event listener might mess up the boolean
state of Q editing or not.

Randall-Jiang committed:
fixing fileclick work not as expected issue

leigaol committed:
fix(amazonq): improve settings description #6382

## Problem
Certain customers are confused while reading the description of
workspace index settings

## Solution
As requested by product, remove the word "open" from the description of
workspace index settings

andrewyuq committed:
feat(amazonq): change active file context limit to 40k #6371

Randall-Jiang committed:
addressed comment

Randall-Jiang committed:
add changelog

siakmun-aws committed:
test(amazonq): messages at different remaining iteration counts #6381

## Problem
Previously the logic of showing different messages at different
remaining iteration counts are tested manually.

## Solution
Adding tests to automate the process.

ashishrp-aws committed:
telemetry(amazonq): unit test generation #6386

## Problem
Incorrect handling of errors for 4xx and 5xx on IDE. Proper error
messages for users and telemetry needs to differentiate between service
errors.

## Solution
- Adding 4XX vs 5XX `httpStatusCode` field to `amazonq_utgGenerateTests`
event.
- Improving error handling in unit test generation.

seshubaws committed:
fix(lambda): AppBuilder messages #6362

## Problem
Several errors that were being shown to customers did not have enough
information for them to take action on the problem.

## Solution
Audited all errors and logs shown to customers and updated them to make
them more legible and actionable.

dhasani23 committed:
fix(amazonq): setMaven correctly #6384

## Problem
`setMaven` is being executed unnecessarily.

## Solution
Move the call to happen just before the local build starts.

ctlai95 committed:
feat(amazonq): grouping options for code issues #6330

## Problem
Code issues are grouped by severity and while this is useful in some
cases, it can be hard to navigate.

## Solution
Let the user decide how to group the issues.
Grouping strategies to start off with:
- Severity (existing behavior)
- File Location (issues in the same file are grouped together)

Hweinstock committed:
perf(childprocess): track/monitor health of spawned processes #6304

## Problem
Toolkit/Q slow performance is consistently reported by customers.
Toolkit/Q features can spawn processes, included many long-lived
processes such as LSP servers. The cost of these processes is hidden and
hard to troubleshoot. Any misbehaving process will not be noticed except
in a coarse "Toolkit/Q causes vscode to be slow" manner.

## Solution
- Track all running processes in a map-like data structure. 
- use a `PollingSet` to periodically "monitor" their system usage (%cpu,
memory bytes), logging warning messages when it exceeds a threshold.
- Add developer command to instantiate spawned processes via
`ChildProcess` wrapper to make this easier to test.

jpinkney-aws committed:
refactor(core): Improve httpResourceFetcher API

jpinkney-aws committed:
refactor(core): Make HttpResourceFetcher platform agnostic (#6379)

## Problem
- Resource fetcher currently only works on node. This causes some
implements to use got directly, some use fetch directly, some use
httpResourceFetcher depending on their use case

## Solution
- Make HttpResourceFetcher platform independent so it can run on node +
web
- Move the streaming based HttpResourceFetcher to a node specific folder

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

jpinkney-aws committed:
refactor(core): Remove composite resource fetcher

avi-alpert committed:
revert(amazonq): defaultServiceConfig endpoint change #6392

Revert endpoint change in previous commit:
https://github.com/aws/aws-toolkit-vscode/commit/062d24a8dad43527bf3fd52f80b60fecce876de5

Jurredr committed:
deps(amazonq): bump mynahui to 4.21.5 #6393

## Description
Bumped MynahUI to version 4.21.4 from 4.21.5. This includes two bug
fixes, as described by the changelog commits and by the [MynahUI release
notes](https://github.com/aws/mynah-ui/releases).

jpinkney-aws committed:
Merge pull request #6390 from jpinkney-aws/fetcher

refactor(core): Improve httpResourceFetcher API

Randall-Jiang committed:
Update packages/amazonq/.changes/next-release/Bug Fix-34e877a1-d2ca-4c93-9dee-5dd2019769fe.json

update Changelog to mention the particular command

Co-authored-by: Justin M. Keyes <jmkeyes@amazon.com>

ctlai95 committed:
fix(amazonq): codefix does not do error handling (#6307)

## Problem

Code fix errors are not handled correctly, only shows a generic error
notification and webview is not updated.


## Solution

Surface the server error to the UI

<img width="658" alt="image"
src="https://github.com/user-attachments/assets/6e0e72d1-7de1-415d-a4d4-891d94e6d7ea"
/>



---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

ashishrp-aws committed:
Merge pull request #6383 from Randall-Jiang/master

fix(amazonq): User can view file diff by file click

seshubaws commented on packages/core/src/awsService/appBuilder/explorer/nodes/deployedNode.ts:
```diff
@@ -101,31 +97,6 @@ export async function generateDeployedNode(
-                            )
-                        ),
-                    ]
-                }
```
Why are we removing this connection check?

valerena replied:
All this block of code is to get the Log Group name using SDK v3. 

In theory we could keep this check for IAM connection, but the connection was only being used to set up the v3 client in the following lines, so in reality we don't really need to check for anything because we're not making any other call.

If there's a failure it should throw the error on the lines before this.

avi-alpert committed:
fix(amazonq): enable input after prompt is rejected, prevent iteration limit exceeded (#6395)

## Problem
When a user encounters an error while updating a README, they see a
"Retry" button, which isn't correct because the previous prompt is not
retried. Instead, the user should be able to input a new prompt.

When a user runs out of iterations in a session, the chat allows them to
continue making README update requests, which results in an "Iteration
limit exceeded error".

## Solution
If a user gets an error while updating a README, the chat input is
enabled and the user can try another prompt.

Once the user hits their iteration limit for the session, they are not
able to suggest more changes and have to accept/reject the existing set
of changes. This will reduce the "iteration limit exceeded" errors
encountered by users.


A E2E test is added to verify that prompt errors during README updates
are handled correctly with the relevant error message, and validates
that users can enter a new prompt.


Screenshots:
_Prompt enabled after error updating README_
<img width="520" alt="image"
src="https://github.com/user-attachments/assets/407109d2-3f10-4e8c-9312-ecab57e625b4"
/>

_'Make changes' button hidden once iteration limit is reached_
<img width="506" alt="image"
src="https://github.com/user-attachments/assets/acf960c0-1c43-475c-a478-c95336ab857b"
/>

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

ctlai95 committed:
config(amazonq): lower timeout for auto reviews (#6397)

## Problem

Auto review timeout is too long


## Solution

Decrease the timeout for auto reviews from 10 to 1 minute. 

`/review` will still use 10 minute timeout.


---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

jpinkney-aws committed:
refactor(core): Allow retries while fetching etags (#6402)

## Problem
We currently only retry normal get requests in the httpResourceFetcher

## Solution
Retry etag get requests as well

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

jpinkney-aws committed:
ci(core): Request should not throw on 304 status (#6400)

## Problem
Only 200-299 are treated as "response.ok" responses by fetch. However,
304's (content not modified) are valid response codes as well

## Solution
If the status code is 304 then don't throw

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

hayemaxi committed:
revert: cloud9 related code (#6376)

This code removes nearly all cloud9 codepaths:
- Icons, readme generation, scripts
- ikp3db lambda/sam debugger support
- Cloud9 only codepaths and hacks
- Docs and comments mentioning Cloud9

Includes some follow up items as TODOs in code.

Some items were left in case they are helpful in the future (e.g. for
vscode web or similar). The items should be renamed from Cloud9 if used
again in the future. Some examples are Cloud9 runtime environment checks
(for lambda/sam), `isCloud9()` and properties/text for Cloud9 IDE, some
SAM artifacts, don't start LSP server, etc. Search cloud9 in vscode
after merging this change.

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

---------

Co-authored-by: Justin M. Keyes <jmkeyes@amazon.com>

avi-alpert committed:
fix(amazonq): /doc prompt refusal error message text update (#6408)

## Problem
A phrase is repeated in the error message for prompt refusal when
updating a README in /doc

## Solution
Fix error message for prompt refusal

Note: changelog is not updated since error message updating already
included [in changelog for this
release](https://github.com/aws/aws-toolkit-vscode/commit/bcdfcdd5d87e90c2d7bdd6a3f7a9b0112eb7bc00#diff-6edd3b49931bb5d80cf0b6d048305bb4902587900e26fea567efe8a1bca38decR3)



---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

dhasani23 committed:
fix(amazonq): replace icons with text (#6398)

## Problem

The multiple icons in the top right of our Transformation Hub were not
easily visible or sufficiently clear to users.

## Solution

Remove the icons, replacing them with a description of what the
icon/button does.

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

---------

Co-authored-by: David Hasani <davhasan@amazon.com>

valerena committed:
deps: update "aws-sdk" to v2.1692.0 #6340

## Problem
The version of the AWSK SDK v2 was outdated 1.5 years, so some newer
functionality was not available.

## Solution
Update the version of `aws-sdk` package to the latest version.

The `Response` class in AWS SDK changed the signature of the `nextPage`
method, so some tests that were mocking this method had to be updated
accordingly (changing the response from `undefined` to `null`).

The definition changed from 
```
    nextPage(callback?: (err: E, data: D) => void): Request<D, E>|void;
```
(callback optional, can return `void`)

To 
```
    nextPage(): Request<D, E> | null;
    nextPage(callback: (err: E, data: D) => void): void;
```
(can return `null` if no callback. only when a callback is passed
returns `void`)

Hweinstock committed:
refactor(pollingset): make bad state impossible #6396

## Problem
Right now `PollingSet` exposes both the `add` and the `start` methods.
The `start` adds the item and starts the timer, while the `add` only
adds the item.

This allows a consumer to create a non-empty `PollingSet` where the
timer never started.

## Solution
- override the add method so that a consumer is forced to start the
timer when adding items.

hayemaxi committed:
fix(notifications): dismiss command visible in command palette (#6407)

Gets rid of these


![image](https://github.com/user-attachments/assets/d0afa2c8-c8c5-43d0-82a0-1ea086be5239)

You can still dismiss by clicking the X button next to non-emergency
notifications.

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

jpinkney-aws committed:
fix(core): Could not fetch endpoints (#6418)

## Problem
While starting the release process we found an problem where the
endpoints couldn't be fetched because the resourceFetcher was expecting
a string but the httpResourceFetcher was returning a Resource

## Solution
If its a string then return the contents from the resource fetcher
directly, otherwise consider it as a response and get the text from the
request

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be
squash-merged at release time.
- Your code changes must meet the guidelines in
[CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the
Apache 2.0 license.

Commit:
Release 3.43.0

Commit:
Release 1.44.0

Commit:
Update version to snapshot version: 3.44.0-SNAPSHOT

Commit:
Update version to snapshot version: 1.45.0-SNAPSHOT

Commit:
Merge release into master

nkomonen-amazon committed:
refactor(util): merge withRetries into waitUntil #6387

## Problem:
`withRetries` and `waitUntil` are two different interfaces for very
similar concepts.

## Solution:
- Add `backoff` and `retryOnFail` flags to `waitUntil()`.
    - `waitUntil(…, {retryOnFail:true})` returns `undefined` if it never
      properly resolves.
    - `waitUntil(…, {retryOnFail:false})` throws if it never properly
      resolves.
- Delete `withRetries()`.

valerena committed:
Update wording for changeset

Co-authored-by: Justin M. Keyes <jmkeyes@amazon.com>
